docs/DeveloperGuide.md

```diff
@@ -167,13 +167,51 @@ Classes used by multiple components are in the `seedu.taassist.commons` package.
+
+When deleting classes, all students previously assigned to those classes will have the classes unassigned from them, before the `ModuleClass` objects corresponding to those classes are removed from the collection captured by `TaAssist`.
+
+The following methods in `TaAssist` manages the adding and deleting of classes from the collection:
+
+- `TaAssist#addModuleClass(ModuleClass moduleClass)` - Adds the provided class to the list of classes created.
+- `TaAssist#removeModuleClass(ModuleClass moduleClass)` - Removes the provided class from the list of classes created.
 
 ### Assigning students to a class
 <img src="images/AssignCommandSequenceDiagram.png" width="700" />
 
-DESCRIPTION TODO
+Each student object contains a collection of `StudentModuleData` where classes and the grades the student obtained for the sessions of the classes are stored. When the user tries to assign students to a class, we are essentially creating a new `StudentModuleData` object to be added to the collection for each student.
+
+Given below are the different steps taken that when the user tries to assign students to a class.
+
+Step 1: The user enters the command keyword `assign`, followed by the indices of the students he want to assign classes to, and the names of the classes that he wants to assign to the students. Example: `assign 1 2 3 c/CS1101S`.
+
+Step 2: The program makes use of the `TAAssistParser` to make sense of the keyword, and determine which parser to use to parse the arguments. In this case, the `AssignCommandParser` will be used.
+
+Step 3: The `AssignCommandParser` makes sense of the arguments, and creates an `AssignCommand` object with the student indices and the classes to assign them to.
+
+Step 4: The `AssignCommand` object is executed. The student indices will be used to retrieve the `Student` objects from the list of students captured by the `Model` interface. For each student, the program will create a new `StudentModuleData` object for each class that is not already assigned to the student. The `StudentModuleData` object will only contain the class name and not any session information. The `StudentModuleData` objects created for the student will be added to the student object's collection of `StudentModuleData`.
+
+Step 5: The execution ends and returns a `CommandResult` object containing the success message to be displayed by the GUI to the user.
+
+#### Design considerations:
+
+* Alternative 1 (current choice): Let each student maintain a collection of classes that the student is being assigned to.
+  * Pros: Only captures necessary information, and easier to implement. This structure is also easier to capture session information for the students.
+  * Cons: Will be creating multiple `StudentModuleData` objects for a class when multiple students are assigned to the class. This can possibly cause performance issue due to the number of objects created.
```

Comment:
```suggestion
  * Cons: Will be creating multiple `StudentModuleData` objects for a class when multiple students are assigned to the class. This may cause performance issues from the large number of objects created.
```